timed/authentication.py

```diff
@@ -33,22 +34,34 @@ def get_introspection(self, access_token, id_token, payload):
         return response.json()
 
     def get_userinfo_or_introspection(self, access_token):
-        try:
-            claims = self.cached_request(
-                self.get_userinfo, access_token, "auth.userinfo"
-            )
-        except requests.HTTPError as e:
-            if not (
-                e.response.status_code in [401, 403] and settings.OIDC_CHECK_INTROSPECT
-            ):
-                raise e
-
-            # check introspection if userinfo fails (confidental client)
-            claims = self.cached_request(
-                self.get_introspection, access_token, "auth.introspection"
-            )
-            if "client_id" not in claims:
-                raise SuspiciousOperation("client_id not present in introspection")
+        while True:
+            try:
+                claims = self.cached_request(
+                    self.get_userinfo, access_token, "auth.userinfo"
+                )
+                break
```

Comment:
you could just early-return here
```suggestion
                return claims
```